publication-commons/src/test/java/no/unit/nva/service/impl/DynamoDBPublicatonServiceTest.java

```diff
@@ -0,0 +1,172 @@
+        client = DynamoDBEmbedded.create().amazonDynamoDB();
+        environment = mock(Environment.class);
+        publicationService = new DynamoDBPublicationService(
+                objectMapper,
+                publicationsDynamoDBLocal.getByPublisherIndex()
+        );
+    }
+
+    @Test
+    @DisplayName("calling Constructor When Missing Env Throws Exception")
+    public void callingConstructorWhenMissingEnvThrowsException() {
+        assertThrows(IllegalStateException.class,
+            () -> new DynamoDBPublicationService(client, objectMapper, environment)
+        );
+    }
+
+    @Test
+    @DisplayName("missing Table Env")
+    public void missingTableEnv() {
+        when(environment.get(TABLE_NAME_ENV)).thenReturn(Optional.of(NVA_RESOURCES_TABLE_NAME));
+        assertThrows(IllegalStateException.class,
+            () -> new DynamoDBPublicationService(client, objectMapper, environment)
+        );
+    }
+
+    @Test
+    @DisplayName("missing Index Env")
+    public void missingIndexEnv() {
+        when(environment.get(BY_PUBLISHER_INDEX_NAME_ENV)).thenReturn(Optional.of(BY_PUBLISHER_INDEX_NAME));
+        assertThrows(IllegalStateException.class,
+            () -> new DynamoDBPublicationService(client, objectMapper, environment)
+        );
+    }
+
+    @Test
+    @DisplayName("calling Constructor With All Env")
+    public void callingConstructorWithAllEnv() {
+        Environment environment = Mockito.mock(Environment.class);
+        when(environment.get(DynamoDBPublicationService.TABLE_NAME_ENV)).thenReturn(Optional.of(TABLE_NAME_ENV));
+        when(environment.get(DynamoDBPublicationService.BY_PUBLISHER_INDEX_NAME_ENV))
+                .thenReturn(Optional.of(BY_PUBLISHER_INDEX_NAME));
+        new DynamoDBPublicationService(client, objectMapper, environment);
+    }
+
+    @Test
+    @DisplayName("empty Table Returns No Publications")
+    public void emptyTableReturnsNoPublications() throws IOException, InterruptedException {
+        List<PublicationSummary> publications = publicationService.getPublicationsByOwner(
+                OWNER,
+                PUBLISHER_ID,
+                null);
+
+        assertEquals(0, publications.size());
+    }
+
+    @Test
+    @DisplayName("nonEmpty Table Returns Publications")
+    public void nonEmptyTableReturnsPublications() throws IOException, InterruptedException {
+        Publication publication = publication();
+        insertPublication(publication);
+
+        List<PublicationSummary> publications = publicationService.getPublicationsByOwner(
+                OWNER,
+                PUBLISHER_ID,
+                null);
+
+        assertEquals(1, publications.size());
+        assertEquals(publication.getEntityDescription().getMainTitle(), publications.get(0).getMainTitle());
+        assertEquals(publication.getOwner(), publications.get(0).getOwner());
+
+    }
+
+    @Test
+    @DisplayName("invalid Item Json")
+    public void invalidItemJson() {
```

Comment:
If we return an empty result for an invalid json, how can we distinguish between non existence and invalid data?
I suggest we create an issue for this. How to handle invalid and non-existent data